geode-core/src/main/java/org/apache/geode/internal/util/ListCollectingServiceLoader.java

```diff
@@ -23,17 +21,17 @@
 import java.util.ServiceConfigurationError;
 import java.util.ServiceLoader;
 
+import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import org.apache.geode.annotations.VisibleForTesting;
-import org.apache.geode.internal.logging.LogService;
 
 /**
  * Implements {@link CollectingServiceLoader} by returning a {@link List} of all currently loadable
  * implementations of the given service interface.
  */
 public class ListCollectingServiceLoader<S> implements CollectingServiceLoader<S> {
-  private static final Logger logger = LogService.getLogger();
+  private static final Logger logger = LogManager.getLogger();
```

Comment:
This is more of a question than a concern, but when do we use LogManager vs LogService to get the logger?  I see a mix of both in this PR.  Does it depend on the context in some way?

Reply:
LogService uses ListCollectingServiceLoader to load the Provider impl for LogService. This means that LogService isn't ready when ListCollectingServiceLoader is used. If I leave it as LogService then it goes into recursion.

Basically, there are some classes that are used to initialize logging (such as ListCollectingServiceLoader) which need to either not log themselves or they have to use non-Geode logging. 

Classes such as ListCollectingServiceLoader should maybe move to geode-common and then that module must never have a dependency on geode-core or geode-logging. We might want to discuss that further...